pkg/security/credential.go

```diff
@@ -107,3 +109,74 @@ func (s *Credential) AddSelfCommonName() error {
 	s.CertAllowedCN = append(s.CertAllowedCN, cn)
 	return nil
 }
+
+// ToTLSConfigWithVerify constructs a `*tls.Config` from the CA, certification and key
+// paths, and add verify for CN.
+//
+// If the CA path is empty, returns nil.
+func ToTLSConfigWithVerify(caPath, certPath, keyPath string, verifyCN []string) (*tls.Config, error) {
```

Comment:
What about reusing `NewTLSConfig ` in tidb.
ref: https://github.com/pingcap/tidb/blob/d2393e3e4dd3580ac1c1874ec12c07f7e45989c5/util/security.go#L167

Reply:
Great advice! I will try this pkg.